File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        reverse(s.begin(), s.end());
            int i = 0;
            while (s[i] == ' ') {
                ++i;
            }
            size_t pos = s.find(' ', i);
            if (pos != std::string::npos) {
                return pos - i;
            } else {
                return s.length() - i;
            };
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        i = 1
        c= 0
        w = False
        t = len(s)
        for _ in range(t):
            if s[-i-c] == " ":
                i += 1
                if w:
                    break
            else:
                c+=1
                w = True
        
        return c